### PR TITLE
Add dim checks for internal `embedding_bag` functions

### LINKED ISSUE #85213: Segmentation fault in embedding_bag
### 🐛 Describe the bug

Segmentation fault in `embedding_bag`, `_embedding_bag` and `_embedding_bag_forward_only`.

### Example to reproduce

```python
import torch

tensor_0 = torch.full((8, 8, 0, 0, 0,), 1.5e+300, dtype=torch.float64, requires_grad=False)
tensor_1 = torch.full((8, 8, 0, 0, 0,), 1, dtype=torch.int64, requires_grad=False)
tensor_2 = torch.full((8, 8, 0, 0, 0, 14, 13, 0, 0, 6,), 1, dtype=torch.int64, requires_grad=False)
bool_3 = True
int_4 = 0
bool_5 = True
tensor_6 = torch.full((4,), 1, dtype=torch.int64, requires_grad=False)
bool_7 = True
int_8 = 0
torch.embedding_bag(tensor_0, tensor_1, tensor_2, bool_3, int_4, bool_5, tensor_6, bool_7)

import torch

weight = torch.full((2, 0, 0, 6, 6,), 0, dtype=torch.float64, requires_grad=False)
indices = torch.full((2, 0, 0, 6, 6,), 2, dtype=torch.int64, requires_grad=False)
offsets = torch.full((2, 0, 0, 6, 6, 8, 6, 8, 0, 6, 0, 11, 0, 0, 0,), 65534, dtype=torch.int64, requires_grad=False)
scale_grad_by_freq = True
mode = 0
sparse = True
per_sample_weights = torch.full((1, 1, 1, 1, 1, 1, 1, 1, 1, 1,), 3.5e+35, dtype=torch.float64, requires_grad=False)
include_last_offset = True
padding_idx = 0
torch._embedding_bag(weight, indices, offsets, scale_grad_by_freq, mode, sparse, per_sample_weights, include_last_offset, padding_idx)


import torch

weight = torch.full((2, 0, 0, 6, 6,), 0, dtype=torch.float64, requires_grad=False)
indices = torch.full((2, 0, 0, 6, 6,), 2, dtype=torch.int64, requires_grad=False)
offsets = torch.full((2, 0, 0, 6, 6, 8, 6, 8, 0, 6, 0, 11, 0, 0, 0,), 65534, dtype=torch.int64, requires_grad=False)
scale_grad_by_freq = True
mode = 0
sparse = True
per_sample_weights = torch.full((1, 1, 1, 1, 1, 1, 1, 1, 1, 1,), 3.5e+35, dtype=torch.float64, requires_grad=False)
include_last_offset = True
padding_idx = 0
torch._embedding_bag_forward_only(weight, indices, offsets, scale_grad_by_freq, mode, sparse, per_sample_weights, include_last_offset, padding_idx)
```

### Result
```segmentation fault```

### Expected behavior
Graceful termination or a RuntimeError to be thrown.

### Note
This bug was discovered using fuzz testing.

### Versions

PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 11 (bullseye) (x86_64)
GCC version: (Debian 10.2.1-6) 10.2.1 20210110
Clang version: 11.0.1-2
CMake version: version 3.22.1
Libc version: glibc-2.31

Python version: 3.9.12 (main, Jun  1 2022, 11:38:51)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.10.0-16-amd64-x86_64-with-glibc2.31
Is CUDA available: N/A
CUDA runtime version: Could not collect
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.11.0a0+gitbc2c6ed
[conda] mkl                       2022.0.1           h06a4308_117
[conda] mkl-include               2022.0.1           h06a4308_117
[conda] numpy                     1.22.3           py39h7a5d4dd_0
[conda] numpy-base                1.22.3           py39hb8be1f0_0
[conda] torch                     1.11.0a0+gitbc2c6ed          pypi_0    pypi

cc @ezyang @gchanan @zou3519

### PR DESCRIPTION
**Pull request by kurtamohler**
Fixes #85213

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Comment by kurtamohler** [comment]
The xla test failure is upstream
